### ISSUE TITLE
Ignition & Speed State being outdated. affects trips report as well

### ISSUE THREAD

**Issue by jaimzj** [tananaev/traccar#3231]
Hello,

1) There are occasions where a device drives into a basement parking where there is no GSM/GPRS coverage, thus at the time of network disruption the ignition value and speed value is true and more than 0 respectively. However the device is then in stationary/parked mode with Ignition false and speed 0. But the same will not reflect on traccar until lets say 9 hours later when the device drives out of the basement.

In such case also, the ignition state change will not be recorded (gt06 protcool  / duplicate filters enabled / traccar 3.11) and the trip from 9 hours earlier and the trip after coming out of the basement is recorded as  a single trip.

(The device GT06 was tested for above scenario, and it also has on device storage for network outages, however, when the device came out of basement, it simply continued the trip on traccar and ignition off in the basement and back on 9 hours later was not recorded (here am not sure if device did not send, or device did send and any of the filters ignored it) please do advice. )

2) Similarly in some situaitons the ignition value could be off, however due to GPS drift the speed value last received ends up being more than 1.  (In such case also, although the device is stationary) the speed showing a value more than 0, can there be a solution for the same?

`report.trip.*` these configurations i believe do not address the above scenario, however please do suggest if there is already a solution to this I might be missing out.

Please advice,

**Comment by tananaev** [comment]
You can increase speed threshold to fix second scenario. There is no solution for the first one. Any suggestions on how to fix it?

**Comment by jaimzj** [comment]
Surely I shall try the speed threshold solution 👍 thank you for that. (this configuration is only to identify device movement state right?)

For the first one, Would it be appropriate to cut a trip based on a combination of both (device state offline)  and time difference between previous and latest update.

**Comment by jaimzj** [comment]
Another thing is even when device state online, I can see right now the device is online however ignition state on server is on (and on device = off) this could be because of duplicate filters being enabled?  so above solution might not work. because calculating engine hours depends on ignition state right? and then  I believe with this scenario engine hour calculations also gets impacted where in reality ignition = off. will handing events before filters are applied work?

**Comment by Abyss777** [comment]
How to know if device was in the basement parking? But not in the mountains without GPRS and was really moving for 9 hours?

I would suggest use trackers that has internal archive and send it after connection reestablished. Also I would recommend choose devices that sleep while in stationary state and you would not need filter positions.

**Comment by jaimzj** [comment]
The devices I am using has both internal archive and sleep mode option as well.

During a live trip through mountains without GPRS they work normal with archiving and later updates when back in GPRS.

This scenario I explained above both of it is specific to exactly when the device is parked and states change. (I can disable the filter duplicate and try if you suggest?)

**Comment by Abyss777** [comment]
Yes, I think duplicate filter prevent you get correct data. Try to disable it.

**Comment by jaimzj** [comment]
Sure I shall remove the duplicate filter, and update results here (Might need couple of hours or a day to re-create the whole scenario)

**Comment by jaimzj** [comment]
@Abyss777 I disabled duplicate filter and restarted traccar and when the device came online now, it did update the ignition state to false. I shall re-create the basement scenario and test as well. although speed value still shows 3.9 kn

**Comment by Abyss777** [comment]
Yes, I noticed that some devices do not send last few positions when they stopped. You can try to play with internal device configuration, sleep delay, reporting period or something...

The only thing that could be done from traccar side is increase speed threshold.

**Comment by bsaggarwal1** [comment]
I am exactly facing issue 1 as a result trip doesn't end.

Also the device st902 obd in my case is reporting events/update. 
However traccar is calculating speed as 15km/hr (due to last available gps signal and speed)  before entering basement.
Also this result in additional trips with 0 distance with speed of 15km/hr.
